app/components/concept/index.ts

```diff
@@ -135,11 +118,11 @@ export default class ConceptComponent extends Component<Signature> {
 
   @action
   async handleContinueButtonClick() {
-    if (this.currentBlockGroupIndex === this.allBlockGroups.length - 1) {
+    this.updateLastRevealedBlockGroupIndex(this.currentBlockGroupIndex + 1);
+    this.enqueueConceptEngagementUpdate.perform();
+
+    if (this.currentBlockGroupIndex === this.allBlockGroups.length) {
```

Comment:
@ryan-gang this seems invalid - `currentBlockGroupIndex` shouldn't go out of bounds of `this.allBlockGroups` no?